### PR TITLE
Fix azure Public IP check

### PR DESCRIPTION
**Pull request by AndreasSteinerPF**
## Description
Azure stores the public ip differently due to the primary/secondary architecture.

### CODE REVIEW

File: internal/provider/resource_azure_cloud_router_connection.go [hunk 1]
```diff
@@ -215,7 +215,7 @@ func resourceAzureExpressRouteConnRead(ctx context.Context, d *schema.ResourceDa
 	_ = d.Set("azure_service_key", resp.CloudSettings.AzureServiceKey)
 	_ = d.Set("po_number", resp.PONumber)
 
-	if resp.CloudSettings.PublicIP != "" {
+	if resp.CloudSettings.PublicIP != "" || resp.CloudSettings.PrimaryPublicIP != "" || resp.CloudSettings.SecondaryPublicIP != "" {
 		_ = d.Set("is_public", true)
```
Comment (rjouhann) [inline internal/provider/resource_azure_cloud_router_connection.go:219]: I think we need to set the `primary_public_ip` and `secondary_public_ip` here so those are set in the TF state file.

```
_ = d.Set("primary_public_ip", resp.CloudSettings.PrimaryPublicIP)
_ = d.Set("primary_public_ip", resp.CloudSettings.SecondaryPublicIP)
```

File: internal/packetfabric/cloud_service_aws.go [hunk 2]
```diff
@@ -167,6 +167,8 @@ type CloudSettings struct {
 	PortCrossConnectOcid           string       `json:"port_cross_connect_ocid,omitempty"`
 	PortID                         string       `json:"port_id,omitempty"`
 	PublicIP                       string       `json:"public_ip,omitempty"`
+	PrimaryPublicIP                string       `json:"primary_public_ip,omitempty"`
```
Comment (rjouhann) [inline internal/packetfabric/cloud_service_aws.go:170]: Assuming those 2 attributes will be added in the API? (I don't see them in https://docs.dev.packetfabric.net/api/v2/swagger/#/Cloud%20Router%20Connections/cloud_routers_connection)

Just to confirm the `v1.6.1` release will happen after those attributes will be added in PF.

Comment (AndreasSteinerPF) [reply internal/packetfabric/cloud_service_aws.go:170]: They are already present (and always have been for Azure). It just looks like they're undocumented in the docs.

File: internal/provider/resource_azure_cloud_router_connection.go
Comment (rjouhann) [inline internal/provider/resource_azure_cloud_router_connection.go:1]: Do we want to also add the 2 computed fields in the resource schema after `vlan_id_microsoft` line 121.

```
						"primary_public_ip": {
							Type:     schema.TypeString,
							Computed: true,
						},
						"secondary_public_ip": {
							Type:     schema.TypeString,
							Computed: true,
						},
```

So users can use those to setup BGP in a second step using `packetfabric_cloud_router_bgp_session`

File: internal/provider/resource_azure_cloud_router_connection.go (hunk 1 above)
Comment (vgvm-pf) [reply internal/provider/resource_azure_cloud_router_connection.go:219]: There's also a reference in the function flattenCloudSettings in datasource_cloud_connections.go which might need an update

Comment (rjouhann) [reply internal/provider/resource_azure_cloud_router_connection.go:219]: @vgvm-pf good call, you are correct!

File: internal/provider/resource_azure_cloud_router_connection.go
Comment (AndreasSteinerPF) [reply internal/provider/resource_azure_cloud_router_connection.go:1]: Done

File: internal/provider/resource_azure_cloud_router_connection.go (hunk 1 above)
Comment (AndreasSteinerPF) [reply internal/provider/resource_azure_cloud_router_connection.go:227]: All done, thanks both!

Comment (vgvm-pf) [reply internal/provider/resource_azure_cloud_router_connection.go:227]: Cool and I think also in dataSourceCloudConnections schema in cloud_settings section. Starting to remember this stuff a little

**Review by rjouhann — APPROVED** [review]

**Review by vgvm-pf — COMMENTED** [review]
I think this new changes cover the provider now as well.

**Review by vgvm-pf — APPROVED** [review]

**Pull request merged**